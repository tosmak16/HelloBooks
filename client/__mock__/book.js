const books = [
  {
    id: 1,
    bookId: 1,
    bookTitle: 'THE PEOPLE IN THE TREE',
    author: 'Y. Hayan',
    category: 'Adventure',
    isbn: '345798465',
    stocknumber: '2',
    image: 'l1.jpg',
    summary: 'Lorem Ipronic typesetting, remaininised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum',
  },
  {
    id: 2,
    bookId: 2,
    bookTitle: 'THE LATE PARADE',
    author: 'F. Adam',
    category: 'Adventure',
    isbn: '234560878',
    stocknumber: '0',
    image: 'l2.jpg',
    summary: 'It is a long established fact thaorem Ipsu uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like)',
  },
  {
    id: 3,
    bookId: 3,
    bookTitle: 'I AM THE WOLF',
    author: 'L. Mark',
    category: 'Music',
    isbn: '4678980976',
    stocknumber: '5',
    image: 'l3.jpg',
    summary: 'There are many variations of passages of Lorem Ipsum available, but the majority orem Ipsum generators on the Internet tend to repeat predefined chunks as necessary',

  },
  {
    id: 4,
    bookId: 4,
    bookTitle: 'THE PALE NORTH',
    author: 'H. Clayton',
    category: 'Adventure',
    isbn: '34567887655',
    stocknumber: '3',
    image: 'l4.jpg',
    summary: 'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from ',
  },
  {
    id: 5,
    bookId: 5,
    bookTitle: 'ADULTOLESENCE',
    author: 'G. Hannah',
    category: 'Romance',
    isbn: '45678974567',
    stocknumber: '10',
    image: 'l5.jpg',
    summary: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Urum.'
  },
  {
    id: 6,
    bookId: 6,
    bookTitle: 'WHAT IT MEANS WHEN A MAN FALL FROM SKY',
    author: 'L. Nneka',
    category: 'Adventure',
    isbn: '567244567',
    stocknumber: '10',
    image: 'l6.jpg',
    summary: 'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium?'
  },
  {
    id: 7,
    bookId: 7,
    bookTitle: 'THE SONG AND THE SILENCE',
    author: 'Y. Johnson',
    category: 'Music',
    isbn: '64578974567',
    stocknumber: '10',
    image: 'l7.jpg',
    summary: 'But I must explain to you how all this mists no resultant pleasure  ducimus qui blanditiis praesentium voluptatum culpa qui officia ?'
  },
  {
    id: 8,
    bookId: 8,
    bookTitle: 'THE ILIAD HOMMER',
    author: 'P. Green',
    category: 'Action',
    isbn: '787890974567',
    stocknumber: '8',
    image: 'l8.jpg',
    summary: 'At vero eos et accusamus et iuse delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.'
  },
];

export default books;
