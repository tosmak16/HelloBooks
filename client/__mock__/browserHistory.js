import { createMemoryHistory } from 'react-router';


const browserHistory = createMemoryHistory('/');

export default browserHistory;
