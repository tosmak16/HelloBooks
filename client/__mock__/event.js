export const e = {
  target: {
    name: 'Tosmak',
    value: '10'
  },
};

export class event {
  static preventDefault() {

  }
}

