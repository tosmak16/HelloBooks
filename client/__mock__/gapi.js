const render = () => { };
const signin2 = {
  render,
};
global.gapi = {
  signin2
};
