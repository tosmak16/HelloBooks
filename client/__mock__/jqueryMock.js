import $ from 'jquery';
import React from 'react';

global.$ = global.jQuery = $;

$.prototype.modal = () => { };

global.Materialize = {
  toast: () => { }
};
