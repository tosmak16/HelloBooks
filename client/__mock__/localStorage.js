
export const localStorage = {
  setItem(key, value) {
    return Object.assign(window.localStorage, { [key]: value });
  },
  getItem(key) {
    return window.localStorage[key];
  },
  clear() {
    window.localStorage = {};
  },

  removeItem(key) {
    return delete window.localStorage[key];
  }

};
export default localStorage;
