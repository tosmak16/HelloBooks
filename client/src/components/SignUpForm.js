import React from 'react';
import { Button, Icon, Input, Row } from 'react-materialize';


import map from 'lodash/map';
import PropTypes from 'prop-types';

const membershipType = {
  Basic: 'Basic',
  Silver: 'Silver',
  Gold: 'Gold',
  Bronze: 'Bronze'
};
class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      firstName: '',
      lastName: '',
      email: '',
      membershipType: '',
      errors: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState({ errors: '' });
    this.props.userSignup(this.state);
  }

  render() {
    const { error } = this.props.reg;
    const options = map(membershipType, (val, key) =>
      <option key={ val } value={ val }>{key}</option>);
    return (
      <form onSubmit={ this.handleSubmit } className="form-signin" action="/login">
        <h4 className="form-signin-heading">Create an account</h4>
        {error && <p style={{ color: 'red' }} className="help-block">*{error}*</p>}
        <Row>
          <Input
            placeholder="Firstname"
            id="firstname" name="firstName"
            s={ 12 } label="First Name"
            value={ this.state.firstName }
            onChange={ this.handleChange }
            className="form-control sr-only validate"
            required
          />
        </Row>
        <Row>
          <Input
            placeholder="Last name"
            id="lastname" name="lastName"
            s={ 12 } label="Last Name"
            type="text"
            value={ this.state.lastName }
            onChange={ this.handleChange }
            className="form-control validate"
            required
          />
        </Row>
        <Row>
          <Input
            type="email"
            name="email" s={ 12 }
            value={ this.state.email }
            onChange={ this.handleChange }
            id="inputSignUpEmail"
            className="form-control validate"
            placeholder="Email address"
            required
            label="Email address"
          />
        </Row>

        <Row>
          <Input
            type="text"
            name="username" s={ 12 }
            value={ this.state.username }
            onChange={ this.handleChange }
            id="inputUsername"
            className="form-control validate"
            placeholder="Username"
            required
            label="Username"
          />
        </Row>

        <Row>
          <Input
            type="password"
            id="inputsSignUpPassword"
            name="password" s={ 12 }
            value={ this.state.password }
            onChange={ this.handleChange }
            className="form-control validate"
            placeholder="Password"
            required
            label="Password"
          />
        </Row>

        <Row>
          <Input
            name="membershipType"
            id="inputSignUpselect"
            onChange={ this.handleChange }
            value={ this.state.membershipType }
            required
            s={ 12 } type="select"
          >
            <option defaultValue="" disabled>Membership Type</option>
            {options}
          </Input>
        </Row>

        <a href=""><Button waves="light" id="signUbtn" className="btn btn-lg btn-success btn-block" type="submit">Sign up</Button></a>
      </form >
    );
  }
}

SignUpForm.propTypes = {
  reg: PropTypes.object.isRequired,
  userSignup: PropTypes.func.isRequired,


};

export default SignUpForm;

