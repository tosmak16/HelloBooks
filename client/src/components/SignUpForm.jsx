import React from 'react';
import { Button, Input, Row } from 'react-materialize';

import PropTypes from 'prop-types';
/**
 * @description SignuP Form is a presentational component
 * 
 * @export SignUpForm
 * 
 * @class SignUpForm
 * 
 * @extends {React.Component}
 */
export class SignUpForm extends React.Component {
  /**
   * @param {object} props
   * 
   * @memberof SignUpForm
   */
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      firstName: '',
      lastName: '',
      email: '',
      errors: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  /**
   *@function handleChange
   * @param {object} event
   * @memberof SignUpForm
   * @returns {void}
   */
  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
   * @param {object} event
   * @memberof SignUpForm
   * @returns {void}
   */
  handleSubmit(event) {
    event.preventDefault();
    this.setState({ errors: '' });
    this.props.userSignup(this.state);
  }
  /**
   * @function render
   * 
   * @returns {views} components
   * 
   * @memberof SignUpForm
   */
  render() {
    document.title = 'SignupPage';

    const { error } = this.props.register;
    return (
      <form onSubmit={this.handleSubmit} className="form-signin" action="/login">
        <h4 className="form-signin-heading">Create an account</h4>
        {error && <p style={{ color: 'red' }} className="help-block">*{error}*</p>}

        <Row>
          <Input
            id="firstname"
            name="firstName"
            s={12}
            label="First Name"
            value={this.state.firstName}
            onChange={this.handleChange}
            className="form-control validate"
            required
          />
        </Row>
        <Row>
          <Input
            id="lastname"
            name="lastName"
            s={12}
            label="Last Name"
            type="text"
            value={this.state.lastName}
            onChange={this.handleChange}
            className="form-control validate"
            required
          />
        </Row>

        <Row>
          <Input
            type="email"
            name="email"
            s={12}
            value={this.state.email}
            onChange={this.handleChange}
            id="inputSignUpEmail"
            className="form-control validate"
            required
            label="Email address"
          />
        </Row>

        <Row>
          <Input
            type="text"
            name="username"
            s={12}
            value={this.state.username}
            onChange={this.handleChange}
            id="inputUsername"
            className="form-control validate"
            required
            label="Username"
          />
        </Row>

        <Row>
          <Input
            type="password"
            id="inputsSignUpPassword"
            name="password"
            s={12}
            value={this.state.password}
            onChange={this.handleChange}
            className="form-control validate"
            required
            label="Password"
          />
        </Row>
        <a href=""><Button
          waves="light"
          id="signUbtn"
          className="btn btn-lg btn-success btn-block"
          type="submit"
        >Sign up</Button></a>
      </form >
    );
  }
}
SignUpForm.propTypes = {
  register: PropTypes.objectOf(PropTypes.any).isRequired,
  userSignup: PropTypes.func.isRequired,
};
export default SignUpForm;

