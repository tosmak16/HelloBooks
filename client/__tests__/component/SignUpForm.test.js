import React from 'react';
import { shallow, mount } from 'enzyme';
import render from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';
import { Provider } from 'react-redux';
import expect from 'expect';


import { SignUpForm } from '../../src/components/SignUpForm';
import { e, event } from '../../__mock__/event';

const mockFuction = jest.fn();


describe('Signup Form Component test', () => {
  const initialState = {
    reg: [{
      isRegistered: false,
      data: {},
      error: '',
      isFetching: false,
      response: {}
    }]
  };
  let wrapper;


  it('should be defined', () => {
    expect(SignUpForm).toBeDefined();
  });

  it('should test and take snapshot of SignUpPage', () => {
    const tree = render.create(<SignUpForm userSignup={ mockFuction } reg={{ error: '' }} />);
    expect(tree).toMatchSnapshot();
  });

  it('should test for handle change function', () => {
    wrapper = shallow(<SignUpForm userSignup={ mockFuction } reg={{ error: '' }} />);
    wrapper.instance().handeleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleChange(e);
    wrapper.update();
  });

  it('should test for handle submit function and error state', () => {
    wrapper = shallow(<SignUpForm userSignup={ mockFuction } reg={{ error: 'hello' }} />);
    wrapper.instance().handeleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleSubmit(event);
    wrapper.update();
  });
});
