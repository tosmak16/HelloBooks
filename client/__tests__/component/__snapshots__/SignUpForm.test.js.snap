// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Form Component test should test and take snapshot of SignForm 1`] = `
<form
  action="/login"
  className="form-signin"
  onSubmit={[Function]}
>
  <h4
    className="form-signin-heading"
  >
    Create an account
  </h4>
  
  <div
    className="row"
  >
    <div
      className="col input-field s12"
    >
      <input
        className="form-control validate"
        defaultValue={undefined}
        id="firstname"
        name="firstName"
        onChange={[Function]}
        placeholder={undefined}
        required={true}
        type="text"
        value=""
      />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="firstname"
      >
        First Name
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col input-field s12"
    >
      <input
        className="form-control validate"
        defaultValue={undefined}
        id="lastname"
        name="lastName"
        onChange={[Function]}
        placeholder={undefined}
        required={true}
        type="text"
        value=""
      />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="lastname"
      >
        Last Name
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col input-field s12"
    >
      <input
        className="form-control validate"
        defaultValue={undefined}
        id="inputSignUpEmail"
        name="email"
        onChange={[Function]}
        placeholder={undefined}
        required={true}
        type="email"
        value=""
      />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="inputSignUpEmail"
      >
        Email address
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col input-field s12"
    >
      <input
        className="form-control validate"
        defaultValue={undefined}
        id="inputUsername"
        name="username"
        onChange={[Function]}
        placeholder={undefined}
        required={true}
        type="text"
        value=""
      />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="inputUsername"
      >
        Username
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col input-field s12"
    >
      <input
        className="form-control validate"
        defaultValue={undefined}
        id="inputsSignUpPassword"
        name="password"
        onChange={[Function]}
        placeholder={undefined}
        required={true}
        type="password"
        value=""
      />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="inputsSignUpPassword"
      >
        Password
      </label>
    </div>
  </div>
  <a
    href=""
  >
    <button
      className="btn waves-effect waves-light btn btn-lg btn-success btn-block"
      disabled={false}
      id="signUbtn"
      onClick={undefined}
      type="submit"
    >
      Sign up
    </button>
  </a>
</form>
`;
